core/node/node_sync/src/client.rs

```diff
@@ -96,11 +96,36 @@ impl MainNodeClient for Box<DynClient<L2>> {
     async fn fetch_protocol_version(
         &self,
         protocol_version: ProtocolVersionId,
-    ) -> EnrichedClientResult<Option<api::ProtocolVersion>> {
-        self.get_protocol_version(Some(protocol_version as u16))
+    ) -> EnrichedClientResult<Option<api::ProtocolVersionInfo>> {
+        match self
+            .get_protocol_version_info(Some(protocol_version as u16))
             .rpc_context("fetch_protocol_version")
             .with_arg("protocol_version", &protocol_version)
             .await
+        {
+            Ok(result) => Ok(result),
+            Err(err) => {
```

Comment:
Can probably filter errors here; a vanilla node should return a JSON-RPC error with code `-32601` (method not found). OTOH, I'm not sure whether this error will be retained by proxies, so maybe it does make sense to be more general.

Reply:
Yeah, I considered that but wanted to stay on the safe side just in case. Also, I am logging the error so it's not completely lost.